kythe/cxx/indexer/cxx/KytheVFS.cc

```diff
@@ -121,6 +119,12 @@ llvm::ErrorOr<std::unique_ptr<llvm::vfs::File>> IndexVFS::openFileForRead(
 
 llvm::vfs::directory_iterator IndexVFS::dir_begin(const llvm::Twine& dir,
                                                   std::error_code& error_code) {
+  std::string path = dir.str();
+  if (auto* record =
+          FileRecordForPath(path, BehaviorOnMissing::kReturnError, 0)) {
+    return llvm::vfs::directory_iterator(
+        std::make_shared<DirectoryIteratorImpl>(std::move(path), record));
+  }
```

Comment:
Worth at least logging an error, or is that done in FileRecordForPath?

Reply:
I don't believe it's worth logging an error at this level as it would be very noisy (file not found is a generally expected and unsurprising condition), but it's definitely worth properly reporting the error :-)

Updated and tested.